backend/src/Designer/Controllers/RepositoryController.cs

```diff
@@ -558,8 +559,11 @@ public ActionResult ContentsZip(string org, string repository, [FromQuery] bool
                 return BadRequest("User does not have a local clone of the repository.");
             }
 
-            var outStream = new MemoryStream();
-            using (var archive = new ZipArchive(outStream, ZipArchiveMode.Create, leaveOpen: true))
+            var zipType = full ? "full" : "changes";
+            var zipFileName = $"{org}-{repository}-{zipType}.zip";
+            var zipFilePath = Path.Combine(Path.GetTempPath(), zipFileName);
```

Comment:
Consider grouping all temporary files that the studio creates in one folder. If remember correctly we're already using "altinn" directory under tempPath when setting up RepositoryLocation value in startup.